specutils/tests/test_loaders.py

```diff
@@ -662,27 +662,44 @@ def test_tabular_fits_maskheader(tmp_path):
     disp = np.linspace(1, 1.2, 21) * u.AA
     flux = np.random.normal(0., 1.0e-14, disp.shape[0]) * u.Jy
     hdr = fits.header.Header({'TELESCOP': 'Leviathan', 'APERTURE': 1.8,
-                              'OBSERVER': 'Parsons', 'NAXIS': 1, 'NAXIS1': 8})
+                              'OBSERVER': 'Parsons'})
 
     spectrum = Spectrum1D(flux=flux, spectral_axis=disp, meta={'header': hdr})
     tmpfile = str(tmp_path / '_tst.fits')
     spectrum.write(tmpfile, format='tabular-fits')
 
     # Read it in and check against the original
     with fits.open(tmpfile) as hdulist:
+
+        # Test HDU0 header
         assert hdulist[0].header['NAXIS'] == 0
+        assert hdulist[0].header['OBSERVER'] == 'Parsons'
+
+        # keys relevant to datashape are in HDU1 header
         assert hdulist[1].header['NAXIS'] == 2
         assert hdulist[1].header['NAXIS2'] == disp.shape[0]
+
+        # Test legacy behaviour of metadata in HDU1 header
         assert hdulist[1].header['OBSERVER'] == 'Parsons'
 
+
+def test_tabular_fits_update_header(tmp_path):
+    disp = np.linspace(1, 1.2, 21) * u.AA
+    flux = np.random.normal(0., 1.0e-14, disp.shape[0]) * u.erg / (u.s * u.cm**2 * u.AA)
+    hdr = fits.header.Header({'TELESCOP': 'Crystal', 'OBSERVER': 'Cruz'})
+    spec = Spectrum1D(flux=flux, spectral_axis=disp, meta={'header': hdr})
+    tmpfile = str(tmp_path / '_tst2.fits')
+    spec.write(tmpfile, format='tabular-fits')
+    spectrum = Spectrum1D.read(tmpfile, format='tabular-fits')
+
     # Now write with updated header information from spectrum.meta
     spectrum.meta.update({'OBSERVER': 'Rosse', 'EXPTIME': 32.1, 'NAXIS2': 12})
```

Comment:
I've been looking into what needed to be updated in the tests to account for the `store_data_header` option, and began to wonder if this is really testing what we expect it to do. Basically we now have constellations like
```python
spectrum.meta = {'OBSERVER': 'Rosse', 'header': {'TELESCOP': 'Crystal', 'OBSERVER': 'Cruz'}}
```
which is bound to become very confusing, at a minimum. So if it is now decided that we don't want to encourage the "flat" `meta` layout, perhaps we should consider deprecating the `update_header` option altogether.